assets/js/util/partition-report.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Report partitioning utility.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Splits report data into segments for current and comparison ranges.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Array}  report           Report rows.
+ * @param {Object} args             Additional arguments.
+ * @param {number} args.rangeLength Length of an individual range.
+ * @return {Object} Object with keys for `compareRange` and `currentRange`.
+ */
+export const partitionReport = ( report, { rangeLength } ) => {
```

Comment:
While the code for this utility function right now appears to be quite generic, I'm not sure that's really necessary (see also other comment). It's somewhat of a nit-pick, but it may be better to call the argument here for example `dateRangeLength` for more clarity on what it refers to. The return object properties could possibly be something like `dateRangeEntries` and `compareDateRangeEntries`, just throwing some ideas out there.

TLDR I think it's a good idea to name and document this more focused around date ranges since that's what we actually use it for.

Reply:
I've updated the range length argument to be `dateRangeLength` as you suggested – this cleaned up the consuming code as well 👍 

> The return object properties could possibly be something like `dateRangeEntries` and `compareDateRangeEntries`, just throwing some ideas out there.

This seems a bit more verbose than necessary. We should keep the return focused on the particular segment of the report.

We could also simply return an array from the function and leave the naming to the consumer via destructuring:

```js
const [ compare, current ] = partitionReport( data, { dateRangeLength } )
```

I wouldn't be opposed to this, but our use would be more consistent to return an object like it does now.

Reply:
Using an object sounds good.